MIPLIB

mmghannam opened the pull request.

CharJon commented on geco/mips/miplib/loader.py:
```diff
@@ -0,0 +1,71 @@
+MIPLIB_INSTANCE_URL = "https://miplib.zib.de/WebData/instances/"
+
+
+def load_instances(filters={}, instances_csv=None):
```
While we have not general guide for this, I think we should avoid mutable default arguments and use 'None' instead.

mmghannam replied:
agreed. done.

CharJon commented on geco/mips/miplib/loader.py:
```diff
@@ -0,0 +1,71 @@
+import os
+import pandas as pd
+
+INSTANCES_DIR = tempfile.gettempdir() + "/geco/miplib/instances/"
```
If I'm seeing this correctly, the tempdir does not get deleted by python.
I think we could make use of a class as a MIPLIB loader and add a destructor to it, that deltes the tempfile (and all its contents). As we are caching anyways, having a statefull class seems reasonable. What do you think?

mmghannam replied:
Just a suggestion (to have all the options before making a decision), we can avoid using a class by adding a clear_cache() function, this would work right?

CharJon replied:
This would also work. I just think it's unexpected from a users perspective to have files left on the disk after the program finishes (even if they are placed in /tmp).

mmghannam replied:
I changed the implementation to use temp files, they get deleted when the program finishes execution.

mmghannam replied:
Is there still an advantage to using a class over this implementation?

CharJon commented on geco/mips/miplib/loader.py:
```diff
@@ -0,0 +1,71 @@
+
+    for instance in df["Instance  Ins."]:
+        full_instance_name = instance + ".mps.gz"
+        yield (load_instance(full_instance_name))
```
Redundant parantheses?

mmghannam replied:
yup. removed them.

CharJon commented on geco/mips/miplib/loader.py:
```diff
@@ -0,0 +1,71 @@
+import tempfile
+import requests
```
requests is not part of the standard library, is it?
Is 'urllib' able to do the same job, or shall we add this to our environment?

mmghannam replied:
I am very glad you said this, the urllib code is more readable.

CharJon commented on README.md:
```diff
@@ -46,6 +46,28 @@ for model in generate_n(lambda seed: knapsack.yang_instance(5, seed), 10):
+from geco.mips.miplib import loader
+
+generator = loader.load_instances(
+    filters= {"Status  Sta.": "hard"},
```
This should be "easy" instead of "hard"?

mmghannam replied:
yup. fixed it.

mmghannam commented:
I removed the untested functionality, and refactored into a Loader class. Maybe we can merge this and push the version number and add to it in following PRs.
